props-core/src/main/java/sh/props/FileWatchSvc.java

```diff
@@ -121,13 +126,15 @@ public static FileWatchSvc instance() {
   /** Main file-watching logic. */
   @Override
   public void run() {
-    WatchKey key = null;
-    try {
-      while ((key = this.watcher.take()) != null) {
-        WatchKey current = key;
+    while (true) {
+      try (CloseableWatchKey key = takeNextKey()) {
```

Comment:
PMD bug: refactored the code to avoid allocating variables in a while loop and instead wrapping `WatchKey`s as `Closeable`s.